src/tiledimage.js

```diff
@@ -784,186 +817,168 @@ $.extend($.TiledImage.prototype, $.EventSource.prototype, /** @lends OpenSeadrag
-            1 :
-            zeroRatioT;
-
-        levelOpacity    = Math.min( 1, ( renderPixelRatioC - 0.5 ) / 0.5 );
-
-        levelVisibility = optimalRatio / Math.abs(
-            optimalRatio - renderPixelRatioT
-        );
+        if (!this.wrapVertical &&
+            (viewportBR.y < 0 || viewportTL.y > this._worldHeightCurrent)) {
+            return;
+        }
 
-        // Update the level and keep track of 'best' tile to load
-        best = updateLevel(
-            tiledImage,
-            haveDrawn,
-            drawLevel,
-            level,
-            levelOpacity,
-            levelVisibility,
-            viewportTL,
-            viewportBR,
-            currentTime,
-            best
-        );
+        // Calculate viewport rect / bounds
+        if (!this.wrapHorizontal) {
+            viewportTL.x = Math.max(viewportTL.x, 0);
+            viewportBR.x = Math.min(viewportBR.x, this._worldWidthCurrent );
+        }
 
-        // Stop the loop if lower-res tiles would all be covered by
-        // already drawn tiles
-        if (  providesCoverage( tiledImage.coverage, level ) ) {
-            break;
+        if (!this.wrapVertical) {
+            viewportTL.y = Math.max(viewportTL.y, 0);
+            viewportBR.y = Math.min(viewportBR.y, this._worldHeightCurrent);
         }
-    }
 
-    // Perform the actual drawing
-    drawTiles( tiledImage, tiledImage.lastDrawn );
+        var levelsInterval = this._getLevelsInterval();
+        var lowestLevel = levelsInterval.lowestLevel;
+        var highestLevel = levelsInterval.highestLevel;
+        var bestTile = null;
+        var haveDrawn = false;
+        var currentTime = $.now();
+
+        // Update any level that will be drawn
+        for (var level = highestLevel; level >= lowestLevel; level--) {
+            var drawLevel = false;
+
+            //Avoid calculations for draw if we have already drawn this
+            var currentRenderPixelRatio = viewport.deltaPixelsFromPointsNoRotate(
+                this.source.getPixelRatio(level),
+                true
+            ).x * this._scaleSpring.current.value;
+
+            if (level === lowestLevel ||
+                (!haveDrawn && currentRenderPixelRatio >= this.minPixelRatio)) {
+                drawLevel = true;
+                haveDrawn = true;
+            } else if (!haveDrawn) {
+                continue;
+            }
 
-    // Load the new 'best' tile
-    if (best && !best.context2D) {
-        loadTile( tiledImage, best, currentTime );
-        tiledImage._setFullyLoaded(false);
-    } else {
-        tiledImage._setFullyLoaded(true);
-    }
-}
+            //Perform calculations for draw if we haven't drawn this
+            var targetRenderPixelRatio = viewport.deltaPixelsFromPointsNoRotate(
+                this.source.getPixelRatio(level),
+                false
+            ).x * this._scaleSpring.current.value; //TODO: shouldn't that be target.value?
```

Comment:
Using target.value seems like a reasonable conclusion from the code, but I don't understand why we would want to... theoretically drawing a single frame shouldn't take the target into account at all. I guess the point is to anticipate where we expect to be by the time those future tiles arrive?

Have you played with it to see what happens if you change it?